tests/test_vms_platform.py

```diff
@@ -117,6 +117,36 @@ def _ensure_jessie_backports_disabled(self, vm):
         # We expect zero hits, so confirm output is empty string.
         self.assertEqual(results, "")
 
+    def _ensure_keyring_package_exists_and_has_correct_key(self, vm, is_whonix=False):
+        keyring_path = "/etc/apt/trusted.gpg.d/securedrop-keyring.gpg"
+        # in whonix-gw-15, the keyring name gets appended with ~ on install
+        if is_whonix:
+            keyring_path += "~"
+        # apt-key finger doesnt work here due to stdout/terminal
+        cmd = "gpg --no-default-keyring --keyring {} -k".format(keyring_path)
+        stdout, stderr = vm.run(cmd)
+        results = stdout.rstrip().decode("utf-8")
+        fpf_gpg_pub_key_info = ["{}".format(keyring_path)]
+        if is_whonix:
+            fpf_gpg_pub_key_info.append("----------------------------------------------")
+        else:
+            fpf_gpg_pub_key_info.append("---------------------------------------------")
```

Comment:
Different number of hyphens? That's a head-scratcher. I suspect that number is determined by the length of filepath:

```
>>> hyphens = "----------------------------------------------"
>>> len(hyphens)
46
>>> keyring_path = "/etc/apt/trusted.gpg.d/securedrop-keyring.gpg"
>>> len(keyring_path)
45
>>>
```

Indeed, matches test output:

```
FAIL: test_debian_keyring_config (test_vms_platform.SD_VM_Platform_Tests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/user/securedrop-workstation/tests/test_vms_platform.py", line 248, in test_debian_keyring_config
    self._ensure_keyring_package_exists_and_has_correct_key(vm, is_whonix=True)
  File "/home/user/securedrop-workstation/tests/test_vms_platform.py", line 138, in _ensure_keyring_package_exists_and_has_correct_key
    self.assertEqual(fpf_gpg_pub_key_info, results.split('\n'))
AssertionError: Lists differ: ['/et[86 chars]------', 'pub   rsa4096 2016-10-20 [SC] [expir[219 chars]ss>'] != ['/et[86 chars]-----', 'pub   rsa4096 2016-10-20 [SC] [expire[218 chars]ss>']

First differing element 1:
'----------------------------------------------'
'---------------------------------------------'

  ['/etc/apt/trusted.gpg.d/securedrop-keyring.gpg',
-  '----------------------------------------------',
?   -

+  '---------------------------------------------',
   'pub   rsa4096 2016-10-20 [SC] [expires: 2021-06-30]',
   '      22245C81E3BAEB4138B36061310F561200F4AD77',
   'uid           [ unknown] SecureDrop Release Signing Key',
   'uid           [ unknown] SecureDrop Release Signing Key '
   '<securedrop-release-key@freedom.press>']

----------------------------------------------------------------------
Ran 70 tests in 256.289s
```

Reply:
There was also some variance to what appears to be a default .gnupg.conf in the Whonix template